src/mapdata.h

```diff
@@ -447,6 +447,8 @@ struct submap {
     trap_id         trp[SEEX][SEEY];  // Trap on each square
     int             rad[SEEX][SEEY];  // Irradiation of each square
 
+    bool            known[SEEX][SEEY]; // has the player ever seen this tile
```

Comment:
I stand corrected, that avoids the multiplayer problem completely.

Losing that knowledge through save & reload might be unexpected for the player, but it's probably acceptable.